src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -693,6 +693,69 @@ class spell_rog_pickpocket : public SpellScript
     }
 };
 
+enum vanish
+{
+    SPELL_PARALYZE = 38132,
+    SPELL_CLEAN_ESCAPE_AURA = 23582,
+    SPELL_CLEAN_ESCAPE_HEAL = 23583
+};
+
+// 18461 - Vanish Purge (Server Side)
+class spell_rog_vanish_purge : public SpellScript
+{
+    PrepareSpellScript(spell_rog_vanish_purge);
+
+    bool Validate(SpellInfo const* /*spellInfo*/)
+    {
+        return ValidateSpellInfo({ SPELL_PARALYZE });
+    }
+
+    void HandleRootRemove(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            if (!caster->HasAura(SPELL_PARALYZE)) // Root from Tainted Core SSC, should not be removed by vanish.
+                caster->RemoveAurasWithMechanic(1 << MECHANIC_ROOT);
+    }
+
+    void HandleSnareRemove(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            caster->RemoveAurasWithMechanic(1 << MECHANIC_SNARE);
+    }
+
+    void Register() override
+    {
+        // Blizzard handles EFFECT_0 as the unroot and EFFECT_1 as unsnare. Hence why they are not done in the same place.
```

Comment:
yes this is true. See https://www.wowhead.com/classic/spell=18461/vanish-purge

In vanilla it existed in DBC, in WotLK they removed it and made it serverside.